### ISSUE TITLE
PublishSingle file can not compile if WindowsPackageType not set

### ISSUE THREAD

**Issue by ghost1372** [microsoft/WindowsAppSDK#4250]
### Describe the bug

 if we change application package type from toolbar
![winui3-csharp-launch-profile](https://github.com/microsoft/WindowsAppSDK/assets/9213496/fa4e62d4-3476-485b-84e5-a8a70b04ec20)
Then we compile a project as a PublishSingleFile we get a error

![309556023-0527bf46-8ff9-4fe0-acd9-4586a7372912](https://github.com/microsoft/WindowsAppSDK/assets/9213496/e37b84fe-d0dc-410f-a325-d90b81f3a95a)

For fixing this error we should add 
`<WindowsPackageType>None</WindowsPackageType>`

### Steps to reproduce the bug

project can not compile as a PublishSingleFile if we use vs toolbar for unpackaged mode.

### Expected behavior

Project should be compiled event we set unpackaged in vs toolbar

### NuGet package version

Windows App SDK 1.5.0: 1.5.240227000

### Packaging type

Unpackaged

### Windows version

Windows 11 version 22H2 (22621, 2022 Update)

### IDE

Visual Studio 2022-preview, Visual Studio 2022

**Comment by Scottj1s** [comment]
@ghost1372 Thanks for your report.  These configuration settings are a bit confusing.  The dropdown in your screenshot only selects the debug launch profile to use, as defined in your Properties\launchSettings.json file.  It has no effect on the build configuration of the project, including publishing.  The configuration settings for publishing are typically stored in your Properties\PublishProfiles\*.pubxml files and in your csproj.

/cc @pratikone

**Comment by ghost1372** [comment]
> @ghost1372 Thanks for your report. These configuration settings are a bit confusing. The dropdown in your screenshot only selects the debug launch profile to use, as defined in your Properties\launchSettings.json file. It has no effect on the build configuration of the project, including publishing. The configuration settings for publishing are typically stored in your Properties\PublishProfiles\*.pubxml files and in your csproj.
> 
> /cc @pratikone

thank you for your answer.
so you mean for publishing our app we need to define `<WindowsPackageType>` in csproj? and we should only use vs toolbar for debug stuff?
i always use a dotnet command for publishing my apps
i did not set `<WindowsPackageType>` in csproj and it does not matter if vs toolbar set to Packaged/UnPackaged, i always get output and my exe can be opened without any issue. is this a bug? or normal behavior? Am I doing the publishing operation correctly? or i should always use `<WindowsPackageType>` for UnPackaged?

`dotnet publish -c Release -r win10-x64 -p:Platform=x64 --self-contained true -p:PublishReadyToRun=true --output publish
`

**Comment by Scottj1s** [comment]
Yes, you need to define WindowsPackageType in either your csproj or your pubxml(s), whichever is appropriate for your structure.  Generally, I would define it in the pubxml(s), since it's a publish-related setting.  But if you're always building PublishSingleFile, which requires WindowsPackageType=None, you could keep things simple and just set it once in your csproj.  dotnet publish is fine (or msbuild /t:publish), as long as your single file output is unpackaged.
